python/src/nanoarrow/schema.py

```diff
@@ -957,14 +988,113 @@ def struct(fields, nullable=True) -> Schema:
     >>> import nanoarrow as na
     >>> na.struct([na.int32()])
     Schema(STRUCT, fields=[Schema(INT32)])
-    >>> na.struct([("col1", na.int32())])
-    Schema(STRUCT, fields=[Schema(INT32, name='col1')])
     >>> na.struct({"col1": na.int32()})
     Schema(STRUCT, fields=[Schema(INT32, name='col1')])
     """
     return Schema(Type.STRUCT, fields=fields, nullable=nullable)
 
 
+def list_of(value_type, nullable=True) -> Schema:
```

Comment:
> In R we have `vctrs::list_of()` but if your brain didn't go there

An average Python programmer like has never heard of `vctrs::list_of()` ;)

Reply:
BTW, for `bool`/`boolean` it's the same thing, you could also use `bool_` instead of switching to `boolean`.

I don't have a strong preference here, because `boolean` is a perfectly fine word in this case. PyArrow uses `bool_`, and the spec (fbs file) speaks about `Bool`, but then Arrow C++ uses `BooleanType` / `boolean()`, so it is already a mixed bag.